### ISSUE TITLE
Error with tabs.sendMessage when no response

### ISSUE THREAD

**Issue by Dreamsorcerer** [mozilla/webextension-polyfill#75]
> A Promise that will be fulfilled with the JSON response object sent by the handler of the message in the content script, **or with no arguments if the content script did not send a response**.

When nothing is returned from the content script, Chrome throws an error. The polyfill should presumably implicitly return something if the content script doesn't, or otherwise suppress this error.

**Comment by LilDevil** [comment]
I can't comment on whether this a bug in the polyfill but I've found that you can make the error go away by adding the following to the end of your message handler.

```
return false;

**Comment by Dreamsorcerer** [comment]
Yes, that's what I've done in the Chrome version. But, the goal of the polyfill is to allow web extensions to run unchanged, hence the bug report that it should do this implicitly.

**Comment by PedroS11** [comment]
I have the same issue with `browser.runtime.sendMessage`. With Firefox, I don't receive the response too as I can see from the issues list, it's still a bug to be fixed

**Comment by dessant** [comment]
There are also other browser bugs around the API, the promise returned by `runtime.sendMessage` will never resolve nor reject if you return `undefined` from the background page message listener in Safari 15, so I always send a `null` response when the reply would be `undefined`, or when there is nothing to return to the caller.

Though these bugs should be fixed in browsers. Developers might type check message responses, so the API behavior should not be changed by the polyfill to hide implementation bugs.
